src/awscdk/awscdk-app-java.ts

```diff
@@ -81,39 +66,35 @@ export class AwsCdkJavaApp extends JavaProject {
     this.mainClass = options.mainClass;
     this.mainPackage = mainPackage.join('.');
     this.mainClassName = mainClassName;
-    this.cdkVersion = options.cdkVersion;
 
-    if (!this.cdkVersion) {
-      throw new Error('Required field cdkVersion is not specified.');
-    }
-
-    const cdkMajorVersion = semver.minVersion(this.cdkVersion)?.major ?? 1;
-    if (cdkMajorVersion >= 2) {
-      throw new Error('CDKv2 is still not supported');
-    }
-
-    this.addCdkDependency('core');
-    this.addCdkDependency(...options.cdkDependencies ?? []);
+    this.cdkDeps = new AwsCdkDepsJava(this, {
+      dependencyType: DependencyType.RUNTIME,
+      ...options,
+    });
 
     this.cdkTasks = new CdkTasks(this);
     this.postCompileTask.spawn(this.cdkTasks.synth);
 
     this.cdkConfig = new CdkConfig(this, {
       app: `mvn exec:java --quiet -Dexec.mainClass=${this.mainClass}`,
+      featureFlags: this.cdkDeps.cdkMajorVersion < 2,
       ...options,
     });
 
-    this.addSample();
+    if ( options.sample ?? true ) {
+      this.addSample();
+    }
   }
 
   /**
    * Adds an AWS CDK module dependencies
    *
-   * @param modules The list of modules to depend on (e.g. "core", "aws-lambda", etc)
+   * @param modules The list of modules to depend on (e.g. "aws-lambda", "aws-iam", etc)
+   * @deprecated In CDK 2.x all modules are available by default. Alpha modules should be added using the standard 'deps'
    */
   public addCdkDependency(...modules: string[]) {
     for (const m of modules) {
-      this.addDependency(`software.amazon.awscdk/${m}@${this.cdkVersion}`);
+      this.cdkDeps.addV1Dependencies(`${AwsCdkDepsJava.PACKAGE_GROUP}/${m}`);
```

Comment:
If this goes through `cdkDeps`, wouldn't it make more sense that the PACKAGE_GROUP part will be implicit?

```suggestion
      this.cdkDeps.addV1Dependencies(m);
```

Reply:
`addV1Dependencies` expects fully qualified while I think the signature of `addCdkDependency` of `AwsCdkJavaAppOptions`  expected non qualified names ( eg `aws-lambda` ).

I'm happy to do that but it'll be a breaking change for users.